cpp/src/arrow/compute/exec/key_hash_test.cc

```diff
@@ -208,7 +219,8 @@ class TestVectorHash {
   }
 };
 
-TEST(VectorHash, Basic) {
+template <typename Type>
+void TestVectorHash() {
```

Comment:
Hmm, AppVeyor (Windows) is failing here:

```
C:/projects/arrow/cpp/src/arrow/compute/exec/key_hash_test.cc(239): error C2904: 'TestVectorHash': name already used for a template in the current scope
C:/projects/arrow/cpp/src/arrow/compute/exec/key_hash_test.cc(36): note: see declaration of 'arrow::compute::TestVectorHash'
[145/356] Building CXX object src\arrow\compute\kernels\CMakeFiles\arrow-compute-aggregate-test.dir\Unity\unity_0_cxx.cxx.obj
```

Maybe it doesn't like having both a method and a class named `TestVectorHash`?

```suggestion
void RunTestVectorHash() {
```

Reply:
OK, let's try this.